src/include/migraphx/raw_data.hpp

```diff
@@ -284,10 +313,19 @@ bool operator==(const T& x, const U& y)
     bool result   = x.empty() and y.empty();
     if(not result and xshape == yshape)
     {
-        visit_all(x, y)([&](auto xview, auto yview) { result = xview == yview; },
-                        [&](auto&& xs, auto&& ys) {
-                            result = std::equal(xs.begin(), xs.end(), ys.begin(), ys.end());
-                        });
+        if(xshape.computable())
```

Comment:
I think the `raw_visit` should do this check. It should do `visit` when the type is computable and just use bytes when it is not.

Reply:
Why would you use raw_visit on a type that's computable? It seems quite counterintuitive for raw_visit to ever have a tensor_view that's something other than `byte` type.

Reply:
If raw_visit only returns tensor_view<byte>(conceptually), then you can have a method like `get_raw` or specialization for `get<byte>` to return the raw bytes. There is no need to use a visit at all. The only reason to have `raw_visit` so it can use the computable type when available and fallback to bytes when its not, so you dont need to repeat this logic everywhere. Perhaps there is a better name.